packages/synapse-react-client/src/components/QueryWrapper/ViewMoreQueryResultsButton.test.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react'
+import { render, screen, waitFor } from '@testing-library/react'
+import { ViewMoreQueryResultsButton } from './ViewMoreQueryResultsButton'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import { QueryContextProvider, QueryContextType } from '../QueryContext'
+import userEvent from '@testing-library/user-event'
+
+function setUp(
+  hasNextPage: boolean,
+  initialLimit: number | undefined,
+  queryCount: number,
+  initialLimitIsApplied: boolean,
+) {
+  const user = userEvent.setup()
+  const mockFetchNextPage = jest.fn()
+  const mockOnRemoveInitialLimit = jest.fn()
+
+  const queryContext: Pick<QueryContextType, 'queryMetadataQueryOptions'> = {
+    queryMetadataQueryOptions: {
+      queryKey: ['query-metadata'],
+      queryFn: jest.fn().mockResolvedValue({
+        queryCount: queryCount,
+        concreteType: 'org.sagebionetworks.repo.model.table.QueryResultBundle',
+      }),
+    },
+  }
+
+  const component = render(
+    <QueryContextProvider queryContext={queryContext as QueryContextType}>
+      <ViewMoreQueryResultsButton
+        hasNextPage={hasNextPage}
+        fetchNextPage={mockFetchNextPage}
+        isFetchingNextPage={false}
+        initialLimit={initialLimit}
+        initialLimitIsApplied={initialLimitIsApplied}
+        onRemoveInitialLimit={mockOnRemoveInitialLimit}
+      />
+    </QueryContextProvider>,
+    { wrapper: createWrapper() },
+  )
+
+  return { user, component, mockFetchNextPage, mockOnRemoveInitialLimit }
+}
+
+describe('ViewMoreQueryResultsButton', () => {
```

Comment:
Tests for new component that handles infinite pagination, mostly based on the original tests in `CardContainer`